sklearn/model_selection/tests/test_validation.py

```diff
@@ -1184,6 +1184,79 @@ def test_learning_curve_with_shuffle():
                               test_scores_batch.mean(axis=1))
 
 
+def test_learning_curve_with_stratify():
+    def max_diff_in_ones_between_strata(X, y, stratify, folds_count=5,
+                                        iters_count=10, random_state=0):
+        ones_counter = []
+
+        class MockCountingEstimator(BaseEstimator):
+            """ Dummy classifier: it stores (in global var for this class)
+            count of non-zero-labeled elements in passed y """
+            def fit(self, X_subset, y_subset=None):
+                cur_ones_count = np.count_nonzero(y_subset)
+                ones_counter.append(cur_ones_count)
+                return self
+
+            def score(self, X, y):
+                return 0
+
+        counting_estimator = MockCountingEstimator()
+        cv = StratifiedKFold(folds_count)
+        train_sizes = np.linspace(1.0 / iters_count, 1.0,
+                                  iters_count)
+
+        learning_curve(counting_estimator, X, y, cv=cv,
+                       train_sizes=train_sizes, shuffle=True,
+                       stratify=stratify,
+                       random_state=random_state)
+
+        # extract max possible diff of ones count between strata
+        # inside each training CV fold
+        max_diff = 0
+        for i in range(folds_count):
```

Comment:
Can we use numpy operations instead of these hard-to-read loops and indexing? I'm finding it very hard up read and am pretty sure that starting with `ones_counter = np.asarray(ones_counter).reshape(folds_count, iters_count)` would fix that.